src/Promotion/Decorator/OrderPromotionsUsageModifierDecorator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Advanced Promotion plugin for Sylius.
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusAdvancedPromotionPlugin\Promotion\Decorator;
+
+use MonsieurBiz\SyliusAdvancedPromotionPlugin\Entity\PromotionCouponsAwareInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Promotion\Modifier\OrderPromotionsUsageModifier;
+use Sylius\Component\Core\Promotion\Modifier\OrderPromotionsUsageModifierInterface;
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
+use Symfony\Component\DependencyInjection\Attribute\AutowireDecorated;
+
+#[AsDecorator('sylius.promotion_usage_modifier')]
+final class OrderPromotionsUsageModifierDecorator implements OrderPromotionsUsageModifierInterface
+{
+    public function __construct(
+        #[AutowireDecorated]
+        private readonly OrderPromotionsUsageModifier $promotionProcessor,
+    ) {
+    }
+
+    public function increment(OrderInterface $order): void
+    {
+        $this->promotionProcessor->increment($order);
+        if (!$order instanceof PromotionCouponsAwareInterface) {
+            return;
+        }
+
+        // Same as Sylius but with multiple coupons
+        // @see Sylius\Component\Core\Promotion\Modifier\OrderPromotionsUsageModifier
+        foreach ($order->getPromotionCoupons() as $promotionCoupon) {
+            if (!$promotionCoupon) {
+                continue;
+            }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏is this test necessary?

Reply:
Yes in the declaration the array values are nullable.
Because when the coupon code does not exist, the value is empty to show the input with the error.
So this case may never happen here, but as the field is `Collection<?PromotionCouponInterface>`
I need to check it's not null to pass the tests.